### PR TITLE
feat: update to latest ziggurat-core, SocketAddr, NodesIndices

### PR DESCRIPTION
**Pull request by kylegranger**
next step in refactoring the crawler metrics.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,8 +22,8 @@ spectre = { git = "https://github.com/niklaslong/spectre", rev = "f1d6698" }
 tabled = "0.10"
 time = "0.3"
 toml = "0.6.0"
-ziggurat-core-metrics = { git = "https://github.com/runziggurat/ziggurat-core", tag = "v0.1.2-zgm" }
-ziggurat-core-crawler = { git = "https://github.com/runziggurat/ziggurat-core", rev = "001c943" }
+ziggurat-core-metrics = { git = "https://github.com/kylegranger/ziggurat-core", rev = "d8e4e0a" }
+ziggurat-core-crawler = { git = "https://github.com/kylegranger/ziggurat-core", rev = "d8e4e0a" }
```
Comment (zeapoz) [inline Cargo.toml:26]: What's the reason for not using the `runziggurat` one?

Comment (asmie) [inline Cargo.toml:26]: Shouldn't point to the main ziggurat-core repository instead of user one?

**Review by zeapoz — COMMENTED** [review]
Left one comment, otherwise LGTM :+1:

File: Cargo.toml (hunk 1 above)
Comment (kylegranger) [reply Cargo.toml]: Fixed!  I had the correct revision, but older path slipped in.

Comment (kylegranger) [reply Cargo.toml]: Fixed!

**Review by zeapoz — APPROVED** [review]

**Review by asmie — APPROVED** [review]

**Pull request merged**